fileuploader-kiwiirc-plugin/components/fileListComponent.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+    <div class="kiwi-filebuffer-outer-container">
+        <div v-if="typeof fileList === 'undefined' || !fileList.length" class="kiwi-filebuffer-status-message">
+            No files have recently been <br>
+            uploaded to this channel {{currentBuffer}}
+        </div>
+        <div v-else class="kiwi-filebuffer-inner-container">
+            <span class="kiwi-filebuffer-status-message">Files recently uploaded to {{currentBuffer}}</span><br><br>
+            <table class="kiwi-filebuffer-table">
+                <tr><th>Nick</th><th>Link</th><th>Time</th></tr>
+                <tr v-for="(file, idx) in fileList" :key="file">
+                    <td class="kiwi-filebuffer-table-td">
+                        {{file.nick}}
+                    </td>
+                    <td class="kiwi-filebuffer-table-td">
+                        <a
+                            :href="file.url"
+                            @click.prevent.stop="loadContent(file.url)"
+                            class="kiwi-filebuffer-anchor"
+                        >
+                            {{getFileName(file.url)}}
+                        </a>
+                    </td>
+                    <td class="kiwi-filebuffer-table-td">
+                        {{file.time}}
+                    </td>
+                </tr>
+            </table>
+        </div>
+    </div>
+</template>
+
+<script>
+
+'kiwi public';
+
+export default {
+    data() {
+        return {
+            fileList: [],
+            settings: kiwi.state.setting('fileuploader'),
+        };
+    },
+    methods: {
+        getFileName(file) {
+            let name = file.split('/')[file.split('/').length-1];
+            if (name.length >= 25) {
+                name = name.substring(0, 18) + '...' + name.substring(name.length - 4);
+            }
+            return name;
+        },
+        loadContent(url) {
+            kiwi.emit('mediaviewer.show', url);
+        },
+        messageHandler() {
+            setTimeout(() => this.sharedFiles(kiwi.state.getActiveBuffer()), 1100)
+        },
+        pruneDups(a) {
+            for(let i = 0; i < a.length; ++i) {
+                for(let j = i + 1; j < a.length; ++j) {
+                    if(a[i].url === a[j].url) a.splice(j, 1)
+                }
+            }
+        },
+        sharedFiles(buffer) {
+            this.fileList = []
+            for(let i = 0; i < buffer.messagesObj.messages.length; i++) {
```

Comment:
Don't use the `messagesObj` internal property directly, use the public `buffer.getMessages()` function

Reply:
done